lib/message_bus.rb

```diff
@@ -554,6 +554,25 @@ def keepalive_interval
     @config[:keepalive_interval] || 60
   end
 
+  # Registers a message filter that allows messages to be filtered from the client.
+  #
+  # @param [Regexp] channel_regexp regular expression to match against a message's channel
+  #
+  # @yieldparam [MessageBus::Message] message published to the channel that matched the regular expression
+  # @yieldreturn [Boolean] whether the message should be published to the client
+  # @return [void]
+  def register_message_filter(channel_regexp, &blk)
```

Comment:
`register_client_message_filter(channel_prefix, &blk)` no need to support full regex here we can just use string.start_with? which is faster.